code/debris/debris.cpp

```diff
@@ -610,6 +610,13 @@ object *debris_create_only(int parent_objnum, int parent_ship_class, int alt_typ
 
 	db->model_instance_num = hull_flag ? model_create_instance(objnum, db->model_num) : -1;
 
+	// ensure that any texture replace on parent ship gets copied to debris --wookieejedi
+	polymodel_instance* pmi_parent = model_get_instance(Objects[parent_objnum].instance);
+	polymodel_instance* pmi_debris = model_get_instance(db->model_instance_num);
+	if (pmi_parent->texture_replace != nullptr) {
+		pmi_debris->texture_replace = std::make_shared<model_texture_replace>(*pmi_parent->texture_replace);
```

Comment:
Based on some research, I think this is actually the wrong way to use a shared pointer.  The `std::make_shared()` function is for creating a `shared_ptr` for the first time, not sharing an existing one.  This would result in two `shared_ptr`s with two reference counts that aren't aware of each other.

All you need to do is assign one `texture_replace` to the other.  This also means no nullptr check is necessary since you aren't dereferencing it.

The two similar uses in the two `ADE_VIRTVAR(Textures` functions should be updated as well.

Reply:
Oh nice. And yeah just tested it with just setting equal and it works as expected, so I've pushed an update to incorporate those requests. Thanks!

Reply:
The shared pointer updates look good, but I noticed another issue that ought to be a breaking bug.  I'm surprised testing passed in spite of it. 🤔

The instance field of `object` refers to the index into Ships[], Weapons[], etc. depending on the type of object.  The `model_get_instance()` function needs the *model instance*, which is a different index.  These two indexes aren't correlated and are usually not the same.  After some consideration, I think they *can* be the same in small, simple missions though.

Use the `object_get_model_instance()` function here instead, which should cover all cases.  (I probably should have called it `object_get_model_instance_index()` instead; I'll refactor it later, unless you feel like also doing that in this PR.)